src/wazuh_db/wdb_parser.c

```diff
@@ -3946,3 +3954,85 @@ int wdb_parse_global_set_agent_labels(wdb_t * wdb, char * input, char * output)
 
     return OS_SUCCESS;
 }
+
+int wdb_parse_global_update_unsynced_agents(wdb_t * wdb, char * input, char * output){
+    const char *error = NULL;
+    int agent_id = 0;
+    cJSON *root = NULL;
+    cJSON *json_agent = NULL;
+    cJSON *json_field = NULL;
+    cJSON *json_label = NULL;
+    cJSON *json_labels = NULL;
+    cJSON *json_key = NULL;
+    cJSON *json_value = NULL;
+    cJSON *json_id = NULL;
+
+    /* 
+    * The cJSON_GetErrorPtr() method is not thread safe, using cJSON_ParseWithOpts() instead,
+    * error indicates where the string caused an error.
+    * The third arguments is TRUE and it will give an error if the input string 
+    * contains data after the JSON command
+    */ 
+    root = cJSON_ParseWithOpts(input, &error, TRUE);
+    if (!root) {
+        mdebug1("Global DB Invalid JSON syntax updating unsynced agents.");
+        mdebug2("Global DB JSON error near: %s", error);
+        snprintf(output, OS_MAXSTR + 1, "err Invalid JSON syntax, near '%.32s'", input);
+        return OS_INVALID;
+
+    } else {
+        /*
+        * Iterating over the array with cJSON_ArrayForEach() 
```

Comment:
Not sure whether this comment and explanation are necessary. This function is already used in many parts of the Wazuh code.
Different from this is the comment above where you are making other developers be careful of changing the `cJSON_ParseWithOpts` function that is thread-safe and is well used here where we are in a function executed by many threads.

Reply:
OK, thank you!